services/onboarding_service.rb

```diff
@@ -7,18 +7,32 @@ module FastlaneCI
   class OnboardingService
     include FastlaneCI::Logging
 
+    # @return [GitRepoConfig]
+    attr_reader :ci_config_repo
+
+    # @return [GitHubProviderCredential]
+    attr_reader :clone_user_provider_credential
+
+    def initialize(ci_config_repo:, clone_user_provider_credential:)
+      @ci_config_repo = ci_config_repo
+      @clone_user_provider_credential = clone_user_provider_credential
+    end
+
     # Verify that fastlane.ci is already set up on this machine.
     # If that's not the case, we have to make sure to trigger the initial clone
     def trigger_initial_ci_setup
       logger.info("No config repo cloned yet, doing that now")
 
       # Trigger the initial clone
+      # TODO: remote this in favour of a different approach
```

Comment:
`remote` -> remove
Could you also comment what you think would be a different/better approach?

Reply:
Oops, yeah - autocomplete ftl (_for the loss_).

I don't like how the clone is done implicitly upon invoking the constructor. It's not apparent creating the object will do a remote call to the GitHub API. I'd prefer to have the constructor not have that responsibility, and make a `clone_project!` method which should be invoked after `new` is called.

This would make the clone a bit more apparent. The downside to this approach is those two methods become tightly coupled with object instantiation - but I think the tradeoff of being more explicit is worth it.

Thoughts?

Reply:
#328 🙄

Reply:
#283 sorry

Reply:
I'm just now getting a chance to test #283 @minuscorp, been a busy day, sorry!

Reply:
Hey! No need to rush, just pointing out to avoid wasting people's time 😁